common-libs/common/src/services/DidDocumentService/LocalKeyVault.ts

```diff
@@ -0,0 +1,46 @@
+import { KeyVault } from './KeyVault'
+import { KeysService } from '../index'
+import * as secp256k1 from 'secp256k1'
+
+/**
+ * KeyStore backed implementation when the seed is known
+ */
+export class LocalKeyVault implements KeyVault {
+  private _keyService: KeysService
+
+  constructor(keyService: KeysService) {
+    this._keyService = keyService
+  }
+
+  primaryPublicKey(): Buffer {
+    const { seed } = this._keyService.decryptSeed()
+    const seedHex = seed.toString('hex')
```

Comment:
Does it really make sense to call `decryptSeed` on every `KeyVault` method call?

For example, prior to that refactoring, `ElemDidDocument._getMyDidConfig` only called `decryptSeed` once; now it is four indirect calls (via keyvault)

Reply:
That's a good question, what was the original thinking around keeping the key encrypted in memory? It does not seem to improve security by much, as both the encrypted key and secret are stored next to each other.

The easiest solution is to keep the seed in clear text in memory to avoid repeated decryptions. A slightly more complex way would be to add an `unlock()/lock()` mechanism to `KeyVault`. 

@DenisPopov15 you also raised this, any preferrences?

Reply:
> what was the original thinking around keeping the key encrypted in memory? It does not seem to improve security by much

What's more, once it is decrypted, it is stored in memory in cleartext anyway until (if) garbage collector collects it.

So with these changes, the number of cleartext seed copies in memory is potentially increased 4x.

Reply:
Are you suggesting we should use the clear text version?

Reply:
That seems like most reasonable solution here. Or maybe there are some other drawbacks?

Reply:
yeap, since anyway 2 parts of secret (encrypteddata and encryption key in memory anyway) - I believe we could cache it there then in open way.
P.S. I believe we should create a ticket to make research and try resolve this situation on our backlog (do not keep seed in open way or encryptedSeed with encryptionKey together in memory), but this is separate topic and will require some time to research